neqo-transport/src/connection.rs

```diff
@@ -162,6 +164,7 @@ pub struct Connection {
     pub(crate) crypto: Crypto,
     pub(crate) acks: AckTracker,
     idle_timeout: Option<Instant>, // TODO(#38)
+    pkt_received_but_no_ae_pkt_sent: bool,
```

Comment:
I wonder if you couldn't use an enum for IdleTimeout:

```rust
enum IdleTimeout {
  Init,
  PacketReceived(Instant),
  AckElicitingPacketSent(Instant),
}
impl Default for IdleTimeout {
  fn default() -> Self {
    IdleTimeout::Init
  }
}
impl IdleTimeout {
  pub fn as_instant(&self) -> Instant {
     map self {
        IdleTimeout::Init => panic!("every connection sends a packet");
        IdleTimeout::PacketReceived(t) | PacketReceived::AckElicitingPacketSent(t) => t,
     }
  }
}
```

Reply:
cool, let's try that.